ddsc/core/download.py

```diff
@@ -544,32 +549,94 @@ def __init__(self, algorithm, expected_hash_value, file_path):
         self.algorithm = algorithm
         self.expected_hash_value = expected_hash_value
         self.file_path = file_path
-        self.status = self.determine_status()
 
     def _get_hash_value(self):
         get_hash_value_func = self.algorithm_to_get_hash_value.get(self.algorithm)
         if get_hash_value_func:
             return get_hash_value_func(self.file_path)
         raise ValueError("Unsupported algorithm {}.".format(self.algorithm))
 
-    def determine_status(self):
-        if self._get_hash_value() == self.expected_hash_value:
-            return self.STATUS_OK
-        else:
-            return self.STATUS_FAILED
+    def is_valid(self):
+        return self._get_hash_value() == self.expected_hash_value
+
+    @staticmethod
+    def get_supported_file_hashes(dds_hashes, file_path):
+        """
+        Returns a list of FileHashes for each dict in dds_hashes.
+        :param dds_hashes: [dict]: list of dicts with 'algorithm', and 'value' keys
+        :param file_path: str: path to file to have hash checked
+        :return: [FileHash]
+        """
+        file_hashes = []
+        for hash_info in dds_hashes:
+            algorithm = hash_info.get('algorithm')
+            hash_value = hash_info.get('value')
+            if algorithm in FileHash.algorithm_to_get_hash_value:
+                file_hashes.append(FileHash(algorithm, hash_value, file_path))
+        return file_hashes
+
+    @staticmethod
+    def separate_valid_and_failed_hashes(file_hashes):
+        """
+        Given a list of file hashes seperate them into a list of valid and a list of failed.
+        :param file_hashes: [FileHash]
+        :return: [FileHash], [FileHash]: valid_file_hashes, failed_file_hashes
+        """
+        valid_file_hashes = []
+        failed_file_hashes = []
+        for file_hash in file_hashes:
+            if file_hash.is_valid():
+                valid_file_hashes.append(file_hash)
+            else:
+                failed_file_hashes.append(file_hash)
+        return valid_file_hashes, failed_file_hashes
+
+
+class FileHashStatus(object):
+    STATUS_OK = "OK"
+    STATUS_CONFLICTED = "CONFLICTED"
```

Comment:
The word `CONFLICTED` to me suggests there's a situation that needs to be resolved. Since we're not recommending any action, I would choose a different word. Specifically in the printed list of statuses after download/verification,`CONFLICTED` makes me think there's a problem with the file and I should re-download or something. I'd suggest changing this to something else like `WARNING` or `OK (1 of 3 matched)`.